portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -37,38 +37,40 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int commentCount = getNumberComments(request);
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
-    List<Map.Entry<String,String>> commentSection = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
-      String name = (String) entity.getProperty("name");
-      String text = (String) entity.getProperty("text");
-      Map.Entry<String,String> nameAndComment=new AbstractMap.SimpleEntry<>(name,text);
-      commentSection.add(nameAndComment);
+    class Comment {
+      public void comment(Entity entity, List<List<String>> commentSection) {
+        List<String> info = new ArrayList<String>(); 
+        String name = (String) entity.getProperty("name");
+        String text = (String) entity.getProperty("text");
+        String id = (String) entity.getProperty("id");
+        info.add(name);
+        info.add(text);
+        info.add(id);
+        commentSection.add(info);
+      }
```

Comment:
This isn't exactly what I meant when I was suggesting you add a `Comment` class. What I was suggesting is for you to create a class called `Comment` that defines some `name`, `text` and `id` members. When you serialize an instance of that class with Gson you'll then get a nicely formatted object that looks like

```
{
  "text": "some comment text",
  "id": "4",
  "name": "Some Name"
}
```

This is much nicer to work with on the frontend rather than have to rely on arrays of arrays where the data you want is at some specific array index.